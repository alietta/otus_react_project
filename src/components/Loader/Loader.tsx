import React, { FC } from "react";
import { useTheme, Spinner } from "sancho";

const Loader: FC<> = () => {
  const theme = useTheme();
  return (
    <div
      css={{
        height: "100vh",
        width: "100vw",
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        background: theme.colors.background.tint1,
      }}
    >
      <Spinner />
    </div>
  );
};
export { Loader };
