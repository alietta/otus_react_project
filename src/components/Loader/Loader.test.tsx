import React from "react";
import { mount } from "enzyme";
import { Loader } from "./Loader";

describe("Loader", () => {
  it("renders Loader", () => {
    const element = mount(<Loader />);
    expect(element).toMatchSnapshot();
  });
});
