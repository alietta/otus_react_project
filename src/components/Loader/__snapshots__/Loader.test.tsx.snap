// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders Loader 1`] = `
@keyframes animation-0 {
  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-3 {
  height: 100vh;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f1f3f5;
}

.emotion-2 {
  opacity: 0;
  display: inline-block;
  -webkit-transition: opacity 0.4s cubic-bezier(0.35,0,0.25,1);
  transition: opacity 0.4s cubic-bezier(0.35,0,0.25,1);
}

.emotion-1 {
  color: #212529;
  text-align: center;
  display: inline-block;
}

.emotion-0 {
  display: inline-block;
  vertical-align: text-bottom;
  border: 0.15em solid currentColor;
  border-right-color: transparent;
  border-radius: 50%;
  -webkit-animation: animation-0 0.75s linear infinite;
  animation: animation-0 0.75s linear infinite;
  width: 1rem;
  height: 1rem;
}

<Loader>
  <div
    className="emotion-3"
  >
    <Component>
      <div
        className="Spinner emotion-2"
      >
        <div
          className="Spinner__container emotion-1"
          role="status"
        >
          <div
            className="Spinner__spinner emotion-0"
          />
          <Component>
            <div
              style={
                Object {
                  "border": 0,
                  "clip": "rect(0 0 0 0)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "absolute",
                  "width": "1px",
                }
              }
            >
              Loading
            </div>
          </Component>
        </div>
      </div>
    </Component>
  </div>
</Loader>
`;
